packages/dds/tree/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -912,5 +925,27 @@ type ComposeMarks<T> =
  * Only valid in the context of a compose (i.e., the output context of `baseMarks` is the input context of `newMark`).
  */
 function areRelatedReattaches<T>(baseMark: Reattach<T>, newMark: Reattach<T>): boolean {
-	return newMark.detachedBy === baseMark.detachedBy;
+	const newEither = newMark.lastDetachedBy ?? newMark.detachedBy;
+	const baseEither = baseMark.lastDetachedBy ?? baseMark.detachedBy;
+	return newEither === baseEither;
+}
+
+function getIntention(
+	rev: RevisionTag | undefined,
+	revisionMetadata: RevisionMetadataSource,
+): RevisionTag | undefined {
+	return rev === undefined ? undefined : revisionMetadata.getInfo(rev).intention ?? rev;
+}
+
+function areInverseRevisions(
+	rev1: RevisionTag | undefined,
+	rev2: RevisionTag | undefined,
+	revisionMetadata: RevisionMetadataSource,
+): boolean {
+	if (rev1 === undefined || rev2 === undefined || rev1 === rev2) {
+		return false;
+	}
+	const info1 = revisionMetadata.getInfo(rev1);
+	const info2 = revisionMetadata.getInfo(rev2);
+	return rev1 === info2.intention || rev2 === info1.intention;
```

Comment:
Is this equivalent to `getIntention(rev1, revisionMetadata) === getIntention(rev2, revisionMetadata)`?

Reply:
The only difference `getIntention(rev1, revisionMetadata) === getIntention(rev2, revisionMetadata)` would evaluate to true for cases where `rev1 === rev2` (which is guarded against above, and now that I think about it, doesn't need to be) and for cases where `info2.intention === info1.intention`.

Admitedly, I can't think of a scenario that would put us in a position where `rev1 === rev2` and `info2.intention === info1.intention` acctually occur. I wonder if that might be possible with constrains or partial undo. Either way, the code above (minus the `rev1 === rev2` check I'm going to remove) seems like a more faithful implementation of the semantics implied by the function name.